### DISCUSSION TITLE
불변 객체와 가변 객체의 차이점

### DISCUSSION THREAD

**Discussion by GIVEN53** [2024-woowacourse-study/level-interview#111] [Clean Code]
.

**Comment by J-I-H-O** [comment]
**불변 객체**의 경우, 한 번 생성되면 내부 상태가 변경될 수 없습니다.
객체의 상태가 변경되지 않기 때문에 보다 안전하고 예측 가능하며, 디버깅과 유지보수가 용이합니다.

**가변 객체**의 경우, 생성된 후에 내부 상태를 변경할 수 있습니다. 가변 객체는 상태 변경에 따른 부작용이 발생할 수 있으며, 멀티 쓰레드 환경에서 동시에 접근할 때 주의가 필요합니다.
예상치 못한 상태 변화로 인해 버그가 발생할 가능성이 높고, 디버깅과 유지보수가 어려울 수 있습니다.

**Comment by 3Juhwan** [comment]
불변 객체는 생성된 후에 상태를 변경할 수 없습니다. 내부 상태가 변경되지 않으므로 여러 스레드 간에 안전하게 공유될 수 있습니다. 예측 가능하고 안정적인 동작을 제공하여 유지보수에 용이합니다. 

가변 객체는 생성된 후에도 객체의 상태를 변경할 수 있습니다. 여러 스레드 간에 동시에 상태 변경이 일어날 경우 문제가 발생할 수 있으며, 예기치 않은 부작용이 발생할 수 있습니다.

**Comment by GIVEN53** [comment]
불
[불변성의 장단점](https://github.com/2024-woowacourse-study/level-interview/discussions/17)

가
상태 변경 가능
동시성 문제
